addons/sale/static/src/js/product_discount_field.js

```diff
@@ -21,7 +21,17 @@ export class ProductDiscountField extends FloatField {
     }
 
     onChange(ev) {
-        const x2mList = this.props.record.model.root.data.order_line;
+        const modelRootData = this.props.record.model.root.data;
+        let x2mList = { records: [] };
+        if ("order_line" in modelRootData) {
+           x2mList = modelRootData.order_line;
+        } else {
+            // In case of a custom x2m field (from studio for example), try to find the x2m field via resModel
+            const entry = Object.entries(modelRootData).find(([_key, data]) => data.resModel === 'sale.order');
+            if (entry) {
+                x2mList = modelRootData[entry[0]];
+            }
```

Comment:
IMHO, we could simply say that this widget logic is not available in other apps and skip if 'order_line' is not found :thinking:

Reply:
Well, we tend to implement this kind of custom widget while considering the standard view, with its available fields, data, ...
I fear that we would get other feedbacks/bugs because the widget doesn't work when you don't have this or that field, ...
Taking a clear stance seems safer and easier IMHO, and I'm not sure it matters that much to have this logic in custom views, as we don't know the scope and what they're trying to achieve there.